library/bignum.c

```diff
@@ -439,9 +461,11 @@ int mbedtls_mpi_read_string( mbedtls_mpi *X, int radix, const char *s )
     size_t i, j, slen, n;
     mbedtls_mpi_uint d;
     mbedtls_mpi T;
+    MPI_VALIDATE_RET( X != NULL );
+    MPI_VALIDATE_RET( s != NULL );
 
     if( radix < 2 || radix > 16 )
-        return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
+        return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );;
```

Comment:
Double ";"

Reply:
I think this should be made optional too, as I think invalid value will not result in memory errors.

(Same for the other occurrences too.)

Reply:
I agree that it's not likely to result in memory errors, but still I think it shouldn't be a precondition. I'd therefore like to keep this check.

Reply:
Removed double semicolon.